java/messages/src/main/java/com/nedap/retail/messages/Client.java

```diff
@@ -187,9 +187,84 @@ public StockSummary getErpStockStatus(final String id) {
     }
 
     /**
-     * Article API: update article information
+     * Requests the total number of articles registered in !D Cloud.
      *
-     * @param articles Articles to update or add
+     * @return Total number of articles.
+     */
+    public Long articleQuantity() {
+
+        final WebResource resource = resource("/article/v2/quantity");
+        final Map response = get(resource, Map.class);
+        return (Long) response.get("quantity");
+    }
+
+    /**
+     * Retrieve one or more articles based on GTIN value(s) or barcodes. Optionally specify returned Article properties.
+     * When querying on barcode, the barcode type is ignored. All articles that have a barcode that match (case
+     * insensitive for alphanumeric barcodes) one of the given barcodes is returned.
+     *
+     * @param gtins The GTIN(s) of which article information should be returned. Repeat key-value for retrieving
+     *            multiple GTINs.Mutually exclusive with barcodes[].
+     * @param barcodes The barcode(s) of which article information should be returned. Repeat key-value for retrieving
+     *            multiple barcodes.Mutually exclusive with gtins[].
+     * @param fields Optional. Which fields should be included in the response. Can be any of the Article fields. When
+     *            omitted: all fields will be included in the response. Repeat key-value for retrieving multiple fields.
+     * @return List of articles retrieved.
+     */
+    public Articles articleDetails(final List<String> gtins, final List<String> barcodes, final List<String> fields) {
+        WebResource resource = resource("/article/v2/retrieve");
+
+        for (final String gtin : gtins) {
+            resource = resource.queryParam("gtins[]", gtin);
+        }
+
+        for (final String barcode : barcodes) {
+            resource = resource.queryParam("barcodes[]", barcode);
+        }
+
```

Comment:
I know that Nenad followed examples from Matija and our retrieve endpoint will return gtins if gitns are there and barcodes if they are there, not combination. This way someone might think we support both gtins and barcodes.

Should we split this into two methods `articleDetailsByBarcode` and `articleDetailsByGtin` to be more explicit?